### ISSUE TITLE
[stable/wordpress] Readiness and liveness probe failes with multisite

### ISSUE THREAD

**Issue by andreisergiu98** [helm/charts#13176]
**Is this a request for help?**: No
---

**Is this a BUG REPORT or FEATURE REQUEST?**
bug report

**Version of Helm and Kubernetes**:
kubernetes 1.12.7
helm: 2.13.1

**Which chart**:
Wordpress

**What happened**
With a lot of struggling i managed to get a multisite worpdress up and running, the problem now is with the readiness and liveness probes. When i enable multisite urls get hardcoded into the database (that's what i understood from the internet), thats why it's more difficult to get a multisite wordpress running locally, because you have to change all those urls to localhost, but then it's not going to work online anymore.

So when the readiness or liveness probe kicks in, it sends a request to `/wp-login.php` which wordpress is redirecting to `http://subdomain.domain.com` which is down until the pod gets ready because the ingress is waiting for the pod to start and sends 503. It's an endless loop.

Here is what i enabled for multisite
```php
define( 'WP_ALLOW_MULTISITE', true );
define('MULTISITE', true);
define('SUBDOMAIN_INSTALL', false);
define('DOMAIN_CURRENT_SITE', 'subdomain.domain.com'); this is a placeholder
define('PATH_CURRENT_SITE', '/');
define('SITE_ID_CURRENT_SITE', 1);
define('BLOG_ID_CURRENT_SITE', 1);
define('WP_SITEURL', 'http://subdomain.domain.com/'); this is a placeholder
define('WP_HOME', 'http://'subdomain.domain.com/'); this is a placeholder
```
And my htaccess
```apache
<Directory "/opt/bitnami/wordpress/">
RewriteEngine On
RewriteBase /
RewriteRule ^index\.php$ - [L]

# add a trailing slash to /wp-admin
RewriteRule ^([_0-9a-zA-Z-]+/)?wp-admin$ $1wp-admin/ [R=301,L]

RewriteCond %{REQUEST_FILENAME} -f [OR]
RewriteCond %{REQUEST_FILENAME} -d
RewriteRule ^ - [L]
RewriteRule ^([_0-9a-zA-Z-]+/)?(wp-(content|admin|includes).*) $2 [L]
RewriteRule ^([_0-9a-zA-Z-]+/)?(.*\.php)$ $2 [L]
RewriteRule . index.php [L]
</Directory>

# Askimet plugin
<Directory "/opt/bitnami/wordpress/wp-content/plugins/akismet">
  # Only allow direct access to specific Web-available files.

# Apache 2.2
<IfModule !mod_authz_core.c>
        Order Deny,Allow
        Deny from all
</IfModule>

# Apache 2.4
<IfModule mod_authz_core.c>
        Require all denied
</IfModule>

# Akismet CSS and JS
<FilesMatch "^(form\.js|akismet\.js|akismet\.css)$">
        <IfModule !mod_authz_core.c>
                Allow from all
        </IfModule>

        <IfModule mod_authz_core.c>
                Require all granted
        </IfModule>
</FilesMatch>

# Akismet images
<FilesMatch "^logo-full-2x\.png$">
        <IfModule !mod_authz_core.c>
                Allow from all
        </IfModule>

        <IfModule mod_authz_core.c>
                Require all granted
        </IfModule>
</FilesMatch>
</Directory>
```

I hacked through this by adding a random header to the readiness probe and then i created a rule in the htaccess to return 200ok if the header exists. This is hacky solution to get the pod up and running, but renders the probes useless.

I might have a bad configuration, i'm not sure, but i couldn't find any useful information on the internet, and i'm new to kubernetes.

**What you expected to happen**:
The readiness and liveness probes to succeed with multisite.

**Anything else we need to know**:
I opened this issue #13161 and then closed it because i thought i fixed the problem, but it's kind of the same.

**Comment by andreisergiu98** [comment]
If the issue is not with my configuration then this will be fixed with kubernetes 1.14
https://github.com/kubernetes/kubernetes/issues/73172
https://github.com/kubernetes/kubernetes/pull/75416

**Comment by juan131** [comment]
Hi @andreisergiu98 

I think this error was solved on this PR: https://github.com/helm/charts/pull/9390

Could you please take a look? You can modify the httpHeaders to avoid the redirection.

**Comment by andreisergiu98** [comment]
I enabled the following headers, but it's still following the redirects and ends with 503
```yaml
livenessProbeHeaders:
- name: X-Forwarded-Proto
  value: https
readinessProbeHeaders:
- name: X-Forwarded-Proto
  value: https
```

**Comment by juan131** [comment]
Hi @andreisergiu98 

That httpHeaders fix the issue when you're being redirected from http to https. However, it doesn't fix the issue in your case (you're being redirected to a specific domain by WordPress).

I guess you could substitute the `httpGet` probe with a `tcpSocket` (checking the port is active) probe or an `exec` probe (using a `curl` call to check wether Apache is responding as expected).

@javsalgar @tompizmor do you know what other alternative to use?

**Comment by javsalgar** [comment]
Yes, I guess the problem is because of the redirection to a different domain. Maybe what it makes more sense is to have the tcpSocket test.

**Comment by juan131** [comment]
Did you have a chance to try the `tcpSocket` test @andreisergiu98 ?

**Comment by andreisergiu98** [comment]
Sorry for the late response, i was out of my country. It works with `tcpSocket`.  Thanks!

```yaml
livenessProbe:
  initialDelaySeconds: 120
  periodSeconds: 10
  timeoutSeconds: 5
  failureThreshold: 6
  successThreshold: 1
  httpGet: null
  tcpSocket:
    port: 80
readinessProbe:
  initialDelaySeconds: 30
  periodSeconds: 10
  timeoutSeconds: 5
  failureThreshold: 6
  successThreshold: 1
  httpGet: null
  tcpSocket:
    port: 80
```

**Comment by hilt86** [comment]
@andreisergiu98 can you share your process to get multisite working with Helm please? The steps I've taken are :

1. Enabled liveness and readiness probes as above
2. Enabled allowOverrideNone=true and an empty customHTAccessCM=custom-htaccess configmap
3. Add `define( 'WP_ALLOW_MULTISITE', true );` to wp-config.php
4. enabled the network and updated wp-config.php with details provided by network setup. Database is updated.
5. Deleted and re-created configmap with new values. Deleted pods so they get new configmap.

The result is I get really broken redirects...login link points to http://wp-login.php/ 

I'm using nginx ingress controller like : 

```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: wp-ingress
  annotations:
    kubernetes.io/ingress.class: nginx
    certmanager.k8s.io/cluster-issuer: letsencrypt-prod
    nginx.ingress.kubernetes.io/rewrite-target: /
    # required to upload plugin
    nginx.ingress.kubernetes.io/proxy-body-size: 50m
    # limits access to app from specified IP
    nginx.ingress.kubernetes.io/whitelist-source-range: "1.2.3.3/32"
spec:
  tls:
  - hosts:
    - www.example.com
    secretName: wordpress.tls-local
  rules:
  - host: www.example.com
    http:
      paths:
      - path: /
        backend:
          serviceName: poc-wordpress
          servicePort: 80
```
Any ideas?

**Comment by andreisergiu98** [comment]
@hilt86 Yeah i stumbled across the same issue. If i remember correctly i used the relocate flag. https://wordpress.org/support/article/changing-the-site-url/

> Note: When the RELOCATE flag is set to true, the Site URL will be automatically updated to whatever path you are using to access the login screen. This will get the admin section up and running on the new URL, but it will not correct any other part of the setup. You’ll still need to alter those manually. 

You can also try to manually set the base urls: 
`
define( 'WP_HOME', 'http://example.com' );
define( 'WP_SITEURL', 'http://example.com' );
`

**Comment by bsakweson** [comment]
I noticed `WORDPRESS_CONFIG_EXTRA` environment variable is not being appended to the wp-config.php file even when it is set. Is that by design, if not what I am missing? 

Is there an alternate way to enable multisite using this chart? I do not seem find it in the values.yaml file.

**Comment by juan131** [comment]
Hi @bsakweson 

At this moment we don't provide a way to modify that setting in the **wp-config.php** using environment variables.

We have an opened issue to support any kind of custom configuration to the **wp-config.php** as you can find in the link below:

- https://github.com/bitnami/bitnami-docker-wordpress/issues/167

You can build your own WP image (based on `bitnami/wordpress` image ) adding this feature as it's explained in [this link](https://github.com/bitnami/bitnami-docker-wordpress#extend-this-image).

**Comment by bsakweson** [comment]
I am not provisioning a standalone instance. I am rather automating the whole installation process and that does not cut it for me. I need that flexibility. I noticed the ticket you shared has been on hold since April last year, is there any reason for the long hold? In any case, I went back to you use the original WordPress image and implement my own chart. This chart is not usable for me as I must automate multisite deployment.
